suricata/package.yaml

```diff
@@ -1,33 +1,1229 @@
+    type: string
+    default: tcp://0.0.0.0:0000
 pipelines:
-  import:
-    name: Suricata Import
-    description: Ingest Suricata events into the internal storage of the node.
+  publish-from-file:
+    name: Suricata Publish
+    description: Ingest Suricata events from a fileinto a subscription topic.
+    restart-on-error: 5m
+    disabled: false
+    definition: |
+      // tql2
+      load_file "{{ inputs.eve-path }}", follow=true
+      read_suricata
+      publish "suricata"
+  publish-from-network:
+    name: Suricata Publish
+    description: |
+      Ingest Suricata events from a network source into a subscription topic.
+    restart-on-error: 5m
+    disabled: true
+    definition: |
+      // tql2
+      load_file "{{ inputs.eve-address }}"
+      read_suricata
+      buffer 16Mi, policy="drop"
+      publish "suricata"
+  suricata-flow-to-ocsf:
+    name: Suricata Flow to OCSF Network Activity
+    description: Maps Suricata Flow events to an OCSF Network Activity event.
+    restart-on-error: 5m
+    disabled: false
+    definition: |
+      // tql2
+      subscribe "suricata"
+      where @name == "suricata.flow"
+      this = { event: this }
+      this = {
+        // --- Classification (required) ---
+        activity_id: 6,
+        category_uid: 4,
+        class_uid: 4001,
+        severity_id: 1,
+        type_uid: 400106,
+        // --- Context (required) ---
+        metadata: {
+          log_name: "suricata.flow",
+          version: "v1.3.0",
+          product: {
+            name: "Suricata",
+            vendor_name: "Open Information Security Foundation",
+          },
+        },
+        // --- Context (optional) ---
+        unmapped: event,
+        // --- Occurrence (required) ---
+        time: event.timestamp,
+        // --- Primary (required) ---
+        dst_endpoint: {
+          ip: event.dest_ip,
+          port: event.dest_port,
+        },
+        // --- Primary (recommended) ---
+        src_endpoint: {
+          ip: event.src_ip,
+          port: event.src_port,
+        },
+        traffic: {
+          bytes_in: event.flow.bytes_toclient,
+          bytes_out: event.flow.bytes_toserver,
+          packets_in: event.flow.pkts_toclient,
+          packets_out: event.flow.pkts_toserver,
+          bytes: event.flow.bytes_toclient + event.flow.bytes_toserver,
+          packets: event.flow.pkts_toclient + event.flow.pkts_toserver,
+        },
+        status_id: 99,
+        status: event.flow.state,
+      }
+      drop(
+        unmapped.timestamp,
+        unmapped.dest_ip,
+        unmapped.dest_port,
+        unmapped.src_ip,
+        unmapped.src_port,
+        unmapped.flow.bytes_toclient,
+        unmapped.flow.bytes_toserver,
+        unmapped.flow.pkts_toclient,
+        unmapped.flow.pkts_toserver,
+        unmapped.flow.state,
+      )
+      @name = "ocsf.network_activity"
+      publish "ocsf.network_activity"
+  suricata-dns-to-ocsf:
```

Comment:
This pipeline should probably map either `dns.grouped` or `dns.answers`, as they contain the meat of the data.

Reply:
`dns.grouped` and `dns.answers` can be a list of multiple values. We currently can't iterate through lists using TQL2. (cc @jachris)
Should I use `python` for this?

Also NB: Suricata also has [a `detailed` field](https://docs.suricata.io/en/latest/output/eve/eve-json-format.html#event-type-dns) that is optional.

Reply:
Note also the DNS logging changes in Suricata 8.0+: https://docs.suricata.io/en/latest/upgrade/8.0-dns-logging-changes.html

Reply:
Yes, list processing needs some more time. It's a significant effort.

Reply:
I tried to use the `python` operator to map the `dns.answers` array: 

![Screenshot_20240820_165500](https://github.com/user-attachments/assets/2acc1fdd-8134-4678-8801-b8339e6876a4)

, but, uh... I hit another roadblock :see_no_evil: 

![image](https://github.com/user-attachments/assets/e2863805-1ee8-4031-b51b-3e03f6a2278f)

Reply:
I think we can just leave a TODO comment in the pipeline for now for this.

Reply:
I added a TODO in https://github.com/tenzir/library/pull/5/commits/bd99b5e151cb0b2ce833270557ad81c11f3d3aa7